git-quick-stats

```diff
@@ -141,6 +167,93 @@ function commitsCalendarByAuthor() {
     '
 }
 
+# DESC: Shows a heatmap of commits per hour of each day for the last 30 days
+function commitsHeatmap() {
+    optionPicked "Commit Heatmap for the last $_commit_days days"
+
+    color_for_count() {
+        local n=$1
+        if (( n == 0 )); then
+            echo -n "${COLOR_BRIGHT_YELLOW}"
+        elif (( n < 2 )); then
+            echo -n "${COLOR_GOLD}"
+        elif (( n < 3 )); then
+            echo -n "${COLOR_ORANGE}"
+        elif (( n < 4 )); then
+            echo -n "${COLOR_DARK_ORANGE}"
+        elif (( n < 5 )); then
+            echo -n "${COLOR_RED_ORANGE}"
+        elif (( n < 6 )); then
+            echo -n "${COLOR_RED}"
+        elif (( n < 8 )); then
+            echo -n "${COLOR_DARK_RED}"
+        elif (( n < 10 )); then
+            echo -n "${COLOR_DEEP_RED}"
+        else
+            echo -n "${COLOR_DARKEST_RED}"
+        fi
+    }
+
+    printf "Day | Date-Hours |"
+    local h
+    for h in {0..23}; do
+        printf " %2d" "$h"
+    done
+    echo
+    echo "------------------------------------------------------------------------------------------"
+
+    local i
+    for i in $(seq $((_commit_days-1)) -1 0); do
+        local day=$(date -d "-$i days" +"%Y-%m-%d")
+        if [[ $(date -d "$day" +%u) -gt 5 ]]; then
+            printf "\e[1;30m"
```

Comment:
I think we can swap this over to `tput`, too. I think this would be `$COLOR_BOLD` and `$COLOR_BLACK`.